lib/parser.js

```diff
@@ -73,11 +50,14 @@ class Parser extends EventEmitter {
             } catch (e) {
                 const script = utils.escapeImportKeyword(options.source.script);
 
-                this.ast = espree.parse(script, Object.assign({}, options, {
+                const newAstOptions = {
+                    ...options.ast,
                     loc: true,

```

Comment:
Looks like that this overrides applied only for this branch of code and ignored for first parsing try

Reply:
Yeah, this piece of code is a little cryptic. Because of the call to `utils.escapeImportKeyword`, I can't simplify the try/catch block. Why not call `utils.escapeImportKeyword` all the time? Now that the ast options are always the same, this code could be simplified a lot.

Reply:
Sometimes import keyword may fail parsing, so we do this workaround. However it broke code locations. But I forgot what the actual case for that(

Reply:
Yes, the added 'X' to import offsets everything. The rename could have been 'IMPORT' to avoid offsetting the locations.

Reply:
I was also worried that the regexp `/import/g` could replace things you don't want (e.g. a variable named `importedFood`). Maybe include whitespace at the start and the end: `/\simport\s/g`

Reply:
You are right, but before start to change it, I need remember why we do that)

Reply:
I can see that this part of the code comes from the initial commit in the repo. But there is no test files in that commit that has an `import` keyword so I don't know how it was tested.

Reply:
But there is no need to change it now. Maybe open a new issue for it to investigate?

Reply:
Yes, I completelly agree with you.
Initially this library started together with svelte-intellisense plugin for VSCode, so it may arrived from there. I'm check v3 parser and this workaround is not used anymore, so maybe we can also remove this code block for v2 parser.

Reply:
Ok, good to know. I wondered why it said svelte-intellisense at the start of the changelog. Now it makes sense.